### PR TITLE
User agent improvements for reservation modification

### PR DESCRIPTION
**Pull request by ASypula**
Added coordinator_id to the Reservation class so that the ModifyReservation message can be sent to appropriate coordinator.
Additionally some error logging to situations when exceptions are raised.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,4 +6,5 @@ COPY . /app
 RUN pip install -r requirements.txt
 
 RUN chmod +x src/main.py
-CMD python -u src/main.py
+CMD sleep infinity
```
Comment (WasilewM) [inline Dockerfile:9]: why do we need this cmd (`sleep infinity`)?

**Review by WasilewM — COMMENTED** [review]
Looks good to me +1

File: Dockerfile (hunk 1 above)
Comment (ASypula) [reply Dockerfile:9]: If there is no command at all, the container will start and shut down immediately.

**Comment by ASypula** [comment]
Before merging, please could someone run also on their own the second example? To make sure that it works as expected (and  check also in the logs if all information is being passed correctly)

File: Dockerfile (hunk 1 above)
Comment (WasilewM) [reply Dockerfile:9]: alright, makes perfect sense

**Review by szaryvip — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (WasilewM) [reply Dockerfile:9]: Didn't work on my side, needed to add `ENTRYPOINT` below

**Comment by WasilewM** [comment]
> Before merging, please could someone run also on their own the second example? To make sure that it works as expected (and check also in the logs if all information is being passed correctly)

It does on my

**Review by WasilewM — APPROVED** [review]

**Pull request merged**